### ISSUE TITLE
Cannot get 'this' in <ListView> renderRow function

### ISSUE THREAD

**Issue by ReadingSteiner** [facebook/react-native#967]
I wrote a function called `renderMyRow` to render rows in a `ListView`, but I found it is strange that I can't get `this` in this function.

Then I fixed it by add a `bind(this)` in my code, but i wonder why i cannot get 'this' in my `renderMyRow` ?

here is the code
```javascript
class SearchResult extends Component {

	constructor(props) {
		super(props);
		var dataSource = new ListView.DataSource({
			rowHasChanged: (r1,r2) => r1.guid !== r2.guid
		});
		this.state = {
			dataSource: dataSource.cloneWithRows(this.props.listings)
		};
	}

	render() {
		return (
			<ListView dataSource={this.state.dataSource} renderRow={this.renderMyRow} />
		)
	}

	rowPressed(propertyGuid) {
		var property = this.props.listings.filter(prop => prop.guid === propertyGuid)[0];

		this.props.navigator.push({
			title: 'Property',
			component: PropertyView,
			props: {property: property}
		})
	}

	renderMyRow(rowData, sectionID, rowID) {
		console.log(this); // Output: null
		var price = rowData.price_formatted.split(' ')[0];
		return(
			<TouchableHighlight underlayColor='#DDDDDD' onPress={ () => this.rowPressed(rowData.guid) }>
				<View>
					<View style={styles.rowContainer}>
						<Image style={styles.thumb} source={{uri : rowData.img_url}}/>
						<View style={styles.textContainer}>
							<Text style={styles.price}>£{price}</Text>
							<Text numberOfLines={1}>{rowData.title}</Text>
						</View>
					</View>
					<View style={styles.separator}/>
				</View>
			</TouchableHighlight>
		);
	}
}
```

**Comment by gabro** [comment]
This is intended behavior when using ES6 classes. See https://facebook.github.io/react/blog/2015/01/27/react-v0.13.0-beta-1.html#autobinding

### tl; dr
ES6 `Component` class does not autobind `this` like `React.createClass` does.

**Comment by ReadingSteiner** [comment]
Thanks a lot!

**Comment by egghead27** [comment]
Hello, Iam very newbie with react-native and even React framework. I have the same issue with Renderrow and this...
I read the doc that you gave but i wonder how i can make the onPress call work without this.onPress.**Thank you for your Help!

**Comment by rupamgho123** [comment]
`this.renderMyRow = this.renderMyRow.bind(this);` in constructor
